### ISSUE TITLE
Create dataset loader for MongabayConservationDataset

### ISSUE THREAD

**Issue by SamuelCahyawijaya** [SEACrowd/seacrowd-datahub#63]
Dataloader name: `mongabay/mongabay.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?mongabay

|     Dataset| mongabay |
|-------------|---|
| Description | Conservation dataset that was collected from mongabay.co.id contains topic-classification task (multi-label format) and sentiment classification. The dataset consists of 31 important topics that are commonly found in Indonesian conservation articles or general news, and each article can belong to more than one topic. After gathering topics for each article, each article will be classified into one of author's sentiments (positive, neutral, negative) based on related topics. |
| Subsets | Multi-label, Sentiment-classification |
| Languages | ind |
| License | The Unlicense (unlicense) |
| Homepage | https://huggingface.co/datasets/Datasaur/mongabay-experiment |
| HF URL | https://huggingface.co/datasets/Datasaur/mongabay-experiment |
| Paper URL | https://arxiv.org/pdf/2310.11258.pdf |

**Comment by elyanah-aco** [comment]
#self-assign

**Comment by megasiska86** [comment]
Hi all sorry for late response, I think it's better if I assign myself to create the dataloader for this dataset.
But it's okay too if @elyanah-aco alread self-assign to build it. Please let me know if anyone needs my help or explanation regarding this dataset. Thank you all

**Comment by elyanah-aco** [comment]
Hello @megasiska86, no problem if you want to handle the dataloader for this (I haven't started on it anyways). You can assign yourself now

**Comment by megasiska86** [comment]
thank you @elyanah-aco , anyway how to assign myself here? I can't click `Assignees` column
![Screenshot from 2023-11-22 12-36-13](https://github.com/SEACrowd/seacrowd-datahub/assets/80664749/4f520132-77af-4ccb-bc27-4e9c553a1295)

**Comment by elyanah-aco** [comment]
Please comment "#self-assign" just like I did in first comment

**Comment by megasiska86** [comment]
#self-assign

**Comment by holylovenia** [comment]
Hi, may I know if you are still working on this issue? Please let @holylovenia @SamuelCahyawijaya @sabilmakbar know if you need any help.

**Comment by megasiska86** [comment]
yeah still working on it

**Comment by holylovenia** [comment]
Okay then, @megasiska86. Feel free to let us know if you need any help!

**Comment by sabilmakbar** [comment]
Hi @megasiska86, may I know the progress of this dataloader creation? Since it has passed 2+2 weeks of expected completion of dataloader, I will clear the assignee if no update is received by Monday 12 PM UTC.

**Comment by Enliven26** [comment]
#self-assign

**Comment by Enliven26** [comment]
yes

**Comment by Enliven26** [comment]
There is a problem with the softlabel column in the training data. For "TEXT" schema feature, it is required to specify the possible values for the string. For validation and test data, it is ["negatif", "positif", "netral"]. However, softlabel column value in the traning data is in the form of arrays of integers, which I currently don't know what rule to turn it into the possible 3 values. How can I resolve this issue? @sabilmakbar . I might also need help in the citation.

**Comment by Enliven26** [comment]
Also for multi-label subset, the softlabel in validation and test data is also in the form of array

**Comment by holylovenia** [comment]
> https://arxiv.org/pdf/2310.11258.pdf

Hi @megasiska86, could you please help answer @Enliven26's questions as the dataset owner?

**Comment by Enliven26** [comment]
May I get an update about the question? I apologize since I haven't gotten any time to read the paper to find the rule to convert the softlabel into single-value label. Also, since for tags classification the label is in the form of array, I think "TEXT" schema cant be used (?).

**Comment by holylovenia** [comment]
> > There is a problem with the softlabel column in the training data. For "TEXT" schema feature, it is required to specify the possible values for the string. For validation and test data, it is ["negatif", "positif", "netral"]. However, softlabel column value in the traning data is in the form of array of floats, which I currently don't know what rule to turn it into the possible 3 values. How can I resolve this issue? @sabilmakbar . I might also need help on the citation.
> 
> > Also for multi-label subset, the softlabel in validation and test data is also in the form of array
> 
> Hi @megasiska86, could you please help answer @Enliven26's questions as the dataset owner?
> 
> I also have a related question, @megasiska86. The huggingface dataset doesn't seem to have the labels for the topic classification task, so where do we get them?

Let me try mentioning @megasiska86 again in case she missed it, @Enliven26.

**Comment by megasiska86** [comment]
I apologize for missed it for a long time
thank you for your help @Enliven26

I've updated the homepage of my dataset
can be checked here 
- Mongabay-tags-classification: https://huggingface.co/datasets/Datasaur/Mongabay-tags-classification
- Mongabay-sentiment-classification: https://huggingface.co/datasets/Datasaur/Mongabay-sentiment-classifiction

**Comment by megasiska86** [comment]
> May I get an update about the question? I apologize since I haven't gotten any time to read the paper to find the rule to convert the softlabel into single-value label. Also, since in tags classification subset the label is in the form of array of tags, I think "TEXT" schema cant be used (?).

Okay, I'll try move it to text2text schema. Thank you

**Comment by megasiska86** [comment]
#self-assign

**Comment by megasiska86** [comment]
Hi @holylovenia
I've finished the dataloader, but got this access error when push it, may I ask the write access again? Thank you very much

```
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by holylovenia** [comment]
> Hi @holylovenia I've finished the dataloader, but got this access error when push it, may I ask the write access again? Thank you very much
> 
> ```
> git@github.com: Permission denied (publickey).
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> ```

Hi @megasiska86, can you make a pull request instead of trying to push to `master`? Please take a look at [the guide](https://github.com/SEACrowd/seacrowd-datahub/blob/master/DATALOADER.md) here. Thanks for your contribution, @megasiska86.

On another note, this issue had been `#self-assigned` by @Enliven26 when you `#self-assigned`. For this case, I let it pass because it seems that @Enliven26 hasn't started implementing the dataloader, but next time please only `#self-assign` dataloader issues with no assignee.

**Comment by megasiska86** [comment]
> Hi @megasiska86, can you make a pull request instead of trying to push to master? Please take a look at [the guide](https://github.com/SEACrowd/seacrowd-datahub/blob/master/DATALOADER.md) here. Thanks for your contribution, @megasiska86.

Yeah I've tried to push it to the new branch and will create PR from that branch, but got this permission error. This permission error also comes up when I tried to clone the repository

> On another note, this issue had been #self-assigned by @Enliven26 when @megasiska86 #self-assigned. For this case, I let it pass because it seems that @Enliven26 hasn't started implementing the dataloader, but next time please only #self-assign dataloader issues with no assignee, @megasiska86.

> Sorry for the inconvenience, @Enliven26. 🙏

Really sorry for the inconvenience :pray: . I tried to #self-assign because I thought it would give me access to push to the branch, but it didn't. I also creeated the dataloader to solve issue raised in this comment

> May I get an update about the question? I apologize since I haven't gotten any time to read the paper to find the rule to convert the softlabel into single-value label. Also, since in tags classification subset the label is in the form of array of tags, I think "TEXT" schema cant be used (?).

Again sorry for the inconvenience.

**Comment by holylovenia** [comment]
> > Hi @megasiska86, can you make a pull request instead of trying to push to master? Please take a look at [the guide](https://github.com/SEACrowd/seacrowd-datahub/blob/master/DATALOADER.md) here. Thanks for your contribution, @megasiska86.
> 
> Yeah I've tried to push it to the new branch and will create PR from that branch, but got this permission error. This permission error also comes up when I tried to clone the repository

Have you tried forking (not cloning) the SEACrowd/seacrowd-datahub repo? Here is [a detailed guide on how to fork the repo and submit the dataloader](https://github.com/SEACrowd/seacrowd-datahub/blob/master/DATALOADER.md#2-setup-a-local-version-of-the-seacrowd-repo).

**Comment by megasiska86** [comment]
> > > Hi @megasiska86, can you make a pull request instead of trying to push to master? Please take a look at [the guide](https://github.com/SEACrowd/seacrowd-datahub/blob/master/DATALOADER.md) here. Thanks for your contribution, @megasiska86.
> > 
> > 
> > Yeah I've tried to push it to the new branch and will create PR from that branch, but got this permission error. This permission error also comes up when I tried to clone the repository
> 
> Have you tried forking (not cloning) the SEACrowd/seacrowd-datahub repo? Here is [a detailed guide on how to fork the repo and submit the dataloader](https://github.com/SEACrowd/seacrowd-datahub/blob/master/DATALOADER.md#2-setup-a-local-version-of-the-seacrowd-repo).

I will try it, thank you :pray:
